src/replication.c

```diff
@@ -1275,6 +1279,26 @@ void replconfCommand(client *c) {
                 }
             }
             sdsfreesplitres(filters, filter_count);
+        } else if (!strcasecmp(c->argv[j]->ptr,"pseudo-replica")) {
+           /* REPLCONF PSEUDO-REPLICA is used to set this replica 
+            * into 'pseudo-replica' mode to avoid eviction and expiration.
+            * This is used for sync tools which eviction and expiration may 
+            * cause the data corruption. */
+            long pseudo_replica = 0;
```

Comment:
REPLCONF is used to set the client state of the calling client, not the server state.
this means that `pseudo-replica` should be a config (config.c).
And REPLCONF should have a `pseudo-master` to set the current connection behave like a master.
if we tie them together, it may mean we need to reset the server state when that client drops, which not necessarily desired since it may soon re-connect and we still don't want eviction to kick in.

Reply:
Thanks for your review. I have separated these two logics.